### ISSUE TITLE
Terrible performance in Chrome and IE

### ISSUE THREAD

**Issue by francispelland** [selectize/selectize.js#594]
I am running into some intense performance issues.  I wasn't running into this during initial development, however when I rolled out my tools into prod I started running into these issues.

Per form, I am displaying 9 input fields.  3 of those input fields use selectize. While developing, I was running no more than 5 forms per page.  However now that I launched on prod, I have over 50 forms, since each form is a customer.  Though I intend to add pagination, I am expecting to always have 50 per page.

I've updated my code so that I only load selectize on the currently visible form.  The execution time is now around 1-2 seconds rather than 75 seconds. This is tested on the latest version of Chrome, IE 10, and IE 11. This does not happen in IE9.

Using IE's profiler, I was able to drill down into the call stack to notice the slowest method call was "offsetWidth".
![selectize_slowness](https://cloud.githubusercontent.com/assets/4448538/4632240/e656f74e-53bb-11e4-8cb7-30f3b14b9259.png)

Here is the stack for a bit more information:
![selectize_slowness2](https://cloud.githubusercontent.com/assets/4448538/4632303/59de7cdc-53bc-11e4-9224-f29af35d6f34.png)
![selectize_slowness3](https://cloud.githubusercontent.com/assets/4448538/4632304/59e4cf6a-53bc-11e4-985d-332d18e1e264.png)

**Comment by whather** [comment]
I'm also getting horrible performance with autoGrow in Chrome (super choppy performance when typing). @cvillemure it would make my day if you shared your autoGrow implementation.

Cheers!

**Comment by whather** [comment]
I decided to re-implement the autoGrow myself using this implementation: https://github.com/MartinF/jQuery.Autosize.Input

https://github.com/whather/selectize.js/commit/4451526b36a41c69142da5f6cacffa4c72b603f0

**Comment by apneadiving** [comment]
I guess its due to `measureString` https://github.com/brianreavis/selectize.js/blob/master/dist/js/selectize.js#L353 which creates then removes a dom element **_every single time**_.
The input could be added to the dom to be cached and removed when selectize is removed.

**Comment by mikemaccana** [comment]
Perf is slow and could be improved, but -1 for the 'perf is terrible'. Rude.

**Comment by rubinred** [comment]
Please fix this issue, as performance is very poor right now.

**Comment by marcotisi** [comment]
I've just created a new PR (#1197)  which add a plugin called "better_auto_grow".
This plugin should improve measureString performances by appending a "&lt;test&gt;" element only once to body and use it to measure string width.

**Comment by marcotisi** [comment]
Removed old PR (#1197) and created a new one (#1199) which integrates changes directly into the core.

**Comment by marcotisi** [comment]
Any news about merging this pull request?

**Comment by marcotisi** [comment]
Still no news on this?

**Comment by risadams** [comment]
closing stale issues older than one year.
If this issue was closed in error please message the maintainers.
All issues must include a proper title, description, and examples.

**Comment by TyphousCrane654** [comment]
I have a list with about 2700 options and confirm that the performance in Chrome is still very poor. In Firefox it's working well.
